capsules/src/net/udp/driver.rs

```diff
@@ -0,0 +1,494 @@
+//! UDP userspace interface for transmit and receive.
+//!
+//! Implements a userspace interface for sending and receiving UDP messages.
+//! Also exposes a list of interface addresses to the application (currently
+//! hard-coded.
+
+use core::cell::Cell;
+use core::{cmp, mem};
+use kernel::common::cells::TakeCell;
+use kernel::{AppId, AppSlice, Callback, Driver, Grant, ReturnCode, Shared};
+use net::ipv6::ip_utils::IPAddr;
+use net::udp::udp_recv::{UDPReceiver, UDPRecvClient};
+use net::udp::udp_send::{UDPSendClient, UDPSender};
+
+/// Syscall number
+pub const DRIVER_NUM: usize = 0x30002;
+
+const INTERFACES: [IPAddr; 2] = [
+    IPAddr([
+        0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e,
+        0x0f,
+    ]),
+    IPAddr([
+        0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e,
+        0x1f,
+    ]),
+];
+
+#[derive(Debug)]
+struct IPAddrPort {
```

Comment:
Maybe `UDPEndpoint` instead? the port is a property of the UDP protocol, not IP

Reply:
Agreed, done.